### PR TITLE
1. Added min max functionality to table and historgramnew classes

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
2. Added a few basic test for both classes as well
3. Updated includes and guards in header files
4. Updated license

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
Ignore for the moment. Didn't realize bins were meant to be that way.

File: include/votca/tools/histogramnew.h [hunk 1]
```diff
@@ -50,7 +47,7 @@ class HistogramNew
          * @param min lower bound of interval
          * @param max upper bound of interval
          * @param nbins number of bins
-         */void Initialize(double min, double max, int nbins);
+         */void Initialize(double min, double max, int nbins=100);
```
Comment (junghans) [inline include/votca/tools/histogramnew.h]: Please no default from `nbins`!

File: include/votca/tools/table.h [hunk 2]
```diff
@@ -150,9 +167,6 @@ inline void Table::push_back(double x, double y, char flags)
     _flags[n] = flags;
 }
 
-istream &operator>>(istream &in, Table& t);
```
Comment (junghans) [inline include/votca/tools/table.h]: Are you dropping this?

Comment (JoshuaSBrown) [reply include/votca/tools/table.h]: Yes it doesn't do anything.

Comment (JoshuaSBrown) [reply include/votca/tools/table.h]: The function call is empty.

Comment (junghans) [reply include/votca/tools/table.h]: Ok, I see

**Comment by JoshuaSBrown** [comment]
If this keeps failing the tests it is probably because the tests are using the numerical output from this object. It will likely be different as the previous implementation was slightly wrong, which means the tests will need to be updated.

**Comment by junghans** [comment]
I ran one of the test locally with:
```
ctest -V -R spce_ibi
```
The error message says:
```
# ERROR:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     #
# critical: 'csg_resample --in /Users/junghans/computing/votca/src.git/votca/build/csg-tutorials/spce/ibi/CG-CG.dist.tgt --out CG-CG.dist.tgt_smooth.JIuQs --grid 0.0:0.01:0.9 --comment Created on Fri Apr 13 14:01:30 MDT 2018 by junghans@pn1503399-gentoo
called from resample_target.sh, version 1.5-dev gitid: 396edd9 (dirty)
settings file: '/Users/junghans/computing/votca/src.git/votca/build/csg-tutorials/spce/ibi/settings.xml'
working directory: /Users/junghans/computing/votca/src.git/votca/build/csg-tutorials/spce/ibi/step_000' failed #
# For details see the logfile /Users/junghans/computing/votca/src.git/votca/build/csg-tutorials/spce/ibi/inverse.log
```
The above mentioned `inverse.log` has the details:
```
an error occurred:
Loading from file to table is not implemented
```

So basically some feature is still missing in your new implementation.

**Comment by junghans** [comment]
One can reproduce the error in a much simpler case:
```
$ cat tab1.txt
0  0
1  1
$ csg_resample --in tab1.txt --out tab2.txt --grid 0.0:1:0.1
an error occurred:
Loading from file to table is not implemented
```

**Comment by JoshuaSBrown** [comment]
Just so you know that example fails without the changes too. (Not the test) 
```csg_resample --in table.txt --out table2.txt --grid 0.0:1:0.1
an error occurred:
error in AkimaSpline::Interpolate : vectors x and y have to contain at least 4 points```

**Comment by junghans** [comment]
Ok, I oversimplified! The tables in the IBI runs have 100s of points! Just use a table like:
```
0 0
1 1
2 2
3 3
4 4
5 5
```

**Comment by JoshuaSBrown** [comment]
Alright I'll go off a working example let's see what is different:
```
cat table.txt 
0 0
1 1
2 2
3 3
[vagrant@votca-devel-env fork]$ csg_resample --in table.txt --out table2.txt --grid 0.0:1:4 --nocut
[vagrant@votca-devel-env fork]$ cat table2.txt 
# 
0 0 i
1 1 i
2 2 i
3 3 i
4 4 o
```

**Comment by junghans** [comment]
Ok, now it is failing in `table_combine.pl --sum --op d WT-WT.dist.tgt WT-WT.dist.new`.

**Comment by junghans** [comment]
Ok I found the issue:
```
$ cd csg-tutorials/spce/atomistic
$ gmx grompp
$ csg_stat --top topol.tpr --cg water.xml --options settings.xml --trj conf.gro
```
Look at the newly written `CG-CG.dist.new` is has the wrote mesh in it.

Acording to `settings.xml` is should be 0:0.01:1, but:
```
$ head CG-CG.dist.new 
0 0 i
0.0101 0 i
0.0202 0 i
0.0303 0 i
```

**Comment by junghans** [comment]
@JoshuaSBrown I guess it is coming from here: https://github.com/votca/csg/blob/master/src/tools/csg_stat_imc.cc#L140

**Comment by JoshuaSBrown** [comment]
Hmm, interesting.

**Comment by JoshuaSBrown** [comment]
Let me check where newhistogram is being called else where in VOTCA it is likely similar approaches (e.g. adding an extra step have been added) have been used to get the correct behavior.

**Pull request merged**